drivers/cache/Kconfig.andes

```diff
@@ -0,0 +1,33 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2024 ANDES Technology Inc.
+
+DT_COMPAT_ANDESTECH_L2C := andestech,l2c
+
+config CACHE_ANDES
+bool "ANDES external cache driver"
+	default y
+	depends on SOC_SERIES_ANDES_AE350
+	select CACHE_HAS_DRIVER
+	imply DCACHE_LINE_SIZE_DETECT
+	imply ICACHE_LINE_SIZE_DETECT
```

Comment:
Shouldn't we have here also a dependency on `DT_HAS_...` ?

Reply:
In the andes cache driver, syscon driver is only needed when the L2 cache exists.

For `DCACHE_LINE_SIZE_DETECT`, there are already dependencies:  
`DCACHE_LINE_SIZE_DETECT` --> `DCACHE` --> `CPU_HAS_DCACHE` (depends on -->)

Reply:
@wtlee1995 - the dependencies are good. Actually both are fine. What Carlo is describing is a relatively new (>3.x.x) facility that reduces the number of "switches" required to enable a driver.

Prior to `DT_HAS_FOO` (an "automatic" Kconfig), one needed to set `status = "okay"` _and_ also set `CONFIG_FOO=y`. Now it's only necessary to enable the node in devictree (approximately). So 1 "switch" instead of 2.

Reply:
I understand the benefit of using `DT_HAS_...` which can reduce the number of "switches" to enable the driver.

There is no specified cache DTS node, but rather the Kconfig option `CPU_HAS_DCACHE`. And we think there is already a hidden dependency between `CACHE_ANDES` and `CPU_HAS_DCACHE`, so we don't need to add another dependency in the driver.

As for `SYSCON` and `DCACHE_LINE_SIZE_DETECT`, they are not mandatory options to enable the driver; they are merely enhanced features. Additionally, the Andes cache driver targets for the platform with either (a) only L1 cache or (b) both L1 and L2 cache. Regardless of whether it's (a) or (b), we want to enable `DCACHE_LINE_SIZE_DETECT` by default. And we want to enable `SYSON` only in (b).